lib/model/folder.go

```diff
@@ -225,6 +228,10 @@ func (f *folder) Stop() {
 	f.cancel()
 }
 
+func (f *folder) Stopped() <-chan struct{} {
+	return f.stopped
+}
```

Comment:
I *think*, Stop() could wait for f.stopped to be closed, at which point m.Remove(id) would block until Stop() returns.

Reply:
There is RemoveAndWait in 2.0.3 version of suture.